vcr/verifier/signature_verifier.go

```diff
@@ -0,0 +1,128 @@
+package verifier
+
+import (
+	crypt "crypto"
+	"errors"
+	"fmt"
+	"github.com/nuts-foundation/nuts-node/vcr/credential"
+	"github.com/nuts-foundation/nuts-node/vcr/types"
+	"strings"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/jsonld"
+	"github.com/nuts-foundation/nuts-node/vcr/issuer"
+	"github.com/nuts-foundation/nuts-node/vcr/signature"
+	"github.com/nuts-foundation/nuts-node/vcr/signature/proof"
+	"github.com/nuts-foundation/nuts-node/vdr/resolver"
+)
+
+type signatureVerifier struct {
+	keyResolver   resolver.KeyResolver
+	jsonldManager jsonld.JSONLD
+}
+
+// VerifySignature implements the Proof Verification Algorithm: https://w3c-ccg.github.io/data-integrity-spec/#proof-verification-algorithm
+func (sv *signatureVerifier) VerifySignature(credentialToVerify vc.VerifiableCredential, validateAt *time.Time) error {
+	switch credentialToVerify.Format() {
+	case issuer.JSONLDCredentialFormat:
+		return sv.jsonldProof(credentialToVerify, credentialToVerify.Issuer.String(), validateAt)
+	case issuer.JWTCredentialFormat:
+		return sv.jwtSignature(credentialToVerify.Raw(), credentialToVerify.Issuer.String(), validateAt)
+	default:
+		return errors.New("unsupported credential proof format")
+	}
+}
+
+// verifyVPSignature implements the Proof Verification Algorithm: https://w3c-ccg.github.io/data-integrity-spec/#proof-verification-algorithm
+func (sv *signatureVerifier) verifyVPSignature(presentation vc.VerifiablePresentation, validateAt *time.Time) error {
```

Comment:
that this one unexported, and the VerifySignature exported is weird

Reply:
exported; the signatureVerifier should probably get its own interface, but out of scope for now